pkg/arvo/tests/sys/zuse/crypto/aes.hoon

```diff
@@ -0,0 +1,109 @@
+::  tests for the aes block cipher
+::
+::  test vectors from Appendix C of:
+::  https://csrc.nist.gov/publications/fips/fips197/fips-197.pdf
+::
+/+  *test
+=,  aes:crypto
+::
+|%
+::
+++  do-test-vectors-ecba
+  |=  ves=(list [key=@ux in=@ux out=@ux])
+  ^-  tang
+  ?~  ves  ~
+  =-  (weld - $(ves t.ves))
+  =,  i.ves
+  ;:  weld
+    ::  encrypt
+    %+  expect-eq
+        !>  out
+        !>  `@ux`(~(en ecba key) in)
+    ::  decrypt
+    %+  expect-eq
+        !>  in
+        !>  `@ux`(~(de ecba key) out)
+  ==
+::
+++  test-aes-ecba
+  %-  do-test-vectors-ecba
+  :~
+    :+  0x0
+      0x0
+    0x66e9.4bd4.ef8a.2c3b.884c.fa59.ca34.2b2e
+  ::
+    :+  0x0
+      0x1
+    0x58e2.fcce.fa7e.3061.367f.1d57.a4e7.455a
+  ::
+    :+  0x1.0203.0405.0607.0809.0a0b.0c0d.0e0f
+      0x11.2233.4455.6677.8899.aabb.ccdd.eeff
+    0x69c4.e0d8.6a7b.0430.d8cd.b780.70b4.c55a
+  ==
+::
+++  do-test-vectors-ecbb
+  |=  ves=(list [key=@ux in=@ux out=@ux])
+  ^-  tang
+  ?~  ves  ~
+  =-  (weld - $(ves t.ves))
+  =,  i.ves
+  ;:  weld
+    ::  encrypt
+    %+  expect-eq
+        !>  out
+        !>  `@ux`(~(en ecbb key) in)
+    ::  decrypt
+    %+  expect-eq
+        !>  in
+        !>  `@ux`(~(de ecbb key) out)
+  ==
+::
+++  test-aes-ecbb
+  %-  do-test-vectors-ecbb
+  :~
+    :+  0x0
+      0x0
+    0xaae0.6992.acbf.52a3.e8f4.a96e.c930.0bd7
+  ::
+    :+  0x0
+      0x1
+    0xcd33.b28a.c773.f74b.a00e.d1f3.1257.2435
+  ::
+    :+  0x1.0203.0405.0607.0809.0a0b.0c0d.0e0f.1011.1213.1415.1617
+      0x11.2233.4455.6677.8899.aabb.ccdd.eeff
+    0xdda9.7ca4.864c.dfe0.6eaf.70a0.ec0d.7191
+  ==
+::
+++  do-test-vectors-ecbc
+  |=  ves=(list [key=@ux in=@ux out=@ux])
+  ^-  tang
+  ?~  ves  ~
+  =-  (weld - $(ves t.ves))
+  =,  i.ves
+  ;:  weld
+    ::  encrypt
+    %+  expect-eq
+        !>  out
+        !>  `@ux`(~(en ecbc key) in)
+    ::  decrypt
+    %+  expect-eq
+        !>  in
+        !>  `@ux`(~(de ecbc key) out)
+  ==
+::
+++  test-aes-ecbc
+  %-  do-test-vectors-ecbc
+  :~
+    :+  0x0
+      0x0
+    0xdc95.c078.a240.8989.ad48.a214.9284.2087
+  ::
+    :+  0x0
+      0x1
+    0x530f.8afb.c745.36b9.a963.b4f1.c4cb.738b
+  ::
+    :+  0x1.0203.0405.0607.0809.0a0b.0c0d.0e0f.1011.1213.1415.1617.1819.1a1b.1c1d.1e1f
```

Comment:
Try to avoid going wider than 80 characters. You can multi-line `@ux` literals like so:

```hoon
:+   0x1.0203.0405.0607.0809.0a0b.0c0d.0e0f.
    1011.1213.1415.1617.1819.1a1b.1c1d.1e1f
  etc
etc
```

Reply:
Done -- will standardize on 64 hex digits per line